webapi/src/test/scala/org/knora/webapi/e2e/v2/ResourcesRouteV2R2Spec.scala

```diff
@@ -93,6 +94,32 @@ class ResourcesRouteV2R2Spec extends R2RSpec {
             }
         }
 
+        "perform a resource request for the book 'Reise ins Heilige Land' using the simple schema (specified by an HTTP header)" in {
+
+            Get(s"/v2/resources/${URLEncoder.encode("http://rdfh.ch/2a6221216701", "UTF-8")}").addHeader(new SchemaHeader(RouteUtilV2.SIMPLE_SCHEMA_NAME)) ~> resourcesPath ~> check {
```

Comment:
`addHeader(new SchemaHeader(RouteUtilV2.SIMPLE_SCHEMA_NAME)`

Wouldn't it be better for a test to send the complete header as a string: "x-knora-accept-schema: simple"

In order to make sure that the parsing works correctly.

Reply:
There is no way to do that with Akka-HTTP. A custom header has to be specified as a class. There's a link to the relevant docs in `SchemaHeader.scala`.

Reply:
I think this is because the tests are not actual HTTP requests as the browser would send them.

Reply:
This isn't anything specific to testing, it seems to be inherent in the design of the Akka HTTP client API:

https://doc.akka.io/docs/akka-http/current/common/http-model.html#custom-headers

I didn't find any other way to add a custom header to an HTTP request in Akka.